src/tracer/ideal_age_example.F90

```diff
@@ -340,27 +374,122 @@ subroutine ideal_age_tracer_column_physics(h_old, h_new, ea, eb, fluxes, dt, G,
+        do j=js,je ; do i=is,ie
+          nk = floor(BL_layers(i,j))
+
+          do k=1,nk
+            if (G%mask2dT(i,j) > 0.0) then
+              CS%tr(i,j,k,m) = CS%tr(i,j,k,m) + G%mask2dT(i,j)*dt*Isecs_per_year
+            else
+              CS%tr(i,j,k,m) = CS%land_val(m)
+            endif
+          enddo
+
+          k = MIN(nk+1,nz)
+
+          if (G%mask2dT(i,j) > 0.0) then
+            layer_frac = BL_layers(i,j)-nk
+            layer_frac = 0.9
+            CS%tr(i,j,k,m) = layer_frac * (CS%tr(i,j,k,m) + G%mask2dT(i,j)*dt &
+                             *Isecs_per_year) + (1.-layer_frac) * young_val
+          else
+            CS%tr(i,j,k,m) = CS%land_val(m)
+          endif
+
+
+          do k=nk+2,nz
+          if (G%mask2dT(i,j) > 0.0) then
+            CS%tr(i,j,k,m) = young_val
+          else
+            CS%tr(i,j,k,m) = CS%land_val(m)
+          endif
+          enddo
+       enddo ; enddo
+
+      else  ! use real BL depth
+        do j=js,je ; do i=is,ie
+          do k=1,CS%nkbl
+            if (G%mask2dT(i,j) > 0.0) then
+              CS%tr(i,j,k,m) = CS%tr(i,j,k,m) + G%mask2dT(i,j)*dt*Isecs_per_year
+            else
+              CS%tr(i,j,k,m) = CS%land_val(m)
+            endif
+          enddo
+
+          do k=CS%nkbl+1,nz
+          if (G%mask2dT(i,j) > 0.0) then
+            CS%tr(i,j,k,m) = young_val
+          else
+            CS%tr(i,j,k,m) = CS%land_val(m)
+          endif
+          enddo
+        enddo ; enddo
+
+     endif ! use real BL depth
+
+    else ! if BL residence tracer
+
+      if (CS%use_real_BL_depth) then
+        do j=js,je ; do i=is,ie
+          nk = floor(BL_layers(i,j))
+          do k=1,nk
+            if (G%mask2dT(i,j) > 0.0) then
+              CS%tr(i,j,k,m) = young_val
+            else
+              CS%tr(i,j,k,m) = CS%land_val(m)
+            endif
+          enddo
+
+          k = MIN(nk+1,nz)
+          if (G%mask2dT(i,j) > 0.0) then
+            layer_frac = BL_layers(i,j)-nk
+            CS%tr(i,j,k,m) = (1.-layer_frac) * (CS%tr(i,j,k,m) + G%mask2dT(i,j)*dt &
+                             *Isecs_per_year) + layer_frac * young_val
+          else
+            CS%tr(i,j,k,m) = CS%land_val(m)
+          endif
+
+          do k=nk+2,nz
+            if (G%mask2dT(i,j) > 0.0) then
+              CS%tr(i,j,k,m) = CS%tr(i,j,k,m) + G%mask2dT(i,j)*dt*Isecs_per_year
+            else
+              CS%tr(i,j,k,m) = CS%land_val(m)
+            endif
+          enddo
+        enddo ; enddo
+
+      else ! use real BL depth
+        do k=1,CS%nkbl ; do j=js,je ; do i=is,ie
+          if (G%mask2dT(i,j) > 0.0) then
+            CS%tr(i,j,k,m) = young_val
+          else
+            CS%tr(i,j,k,m) = CS%land_val(m)
+          endif
+        enddo ; enddo ; enddo
+
+        if (CS%tracer_ages(m) .and. (year>=CS%tracer_start_year(m))) then
+        !$OMP parallel do default(none) shared(is,ie,js,je,CS,nz,G,dt,Isecs_per_year,m)
+        do k=CS%nkbl+1,nz ; do j=js,je ; do i=is,ie
```

Comment:
Please indent this do-loop block by 2 points relative to the surrounding if block, to adhere to the MOM6 code standards at github.com/mom-ocean/MOM6/wiki/Code-style-guide.